src/hm-admin/services/hydra/HydraCore.service.ts

```diff
@@ -0,0 +1,59 @@
+
+import {Http} from '@angular/http';
+import {Injectable} from '@angular/core';
+import 'rxjs/add/operator/map';
+import {Observable} from 'rxjs';
+
+@Injectable()
+export class HydraCore {
+
+    constructor(private http: Http) {
+        this.getHydraCore('https://www.w3.org/ns/hydra/core');
+    }
+
+    getHydraCore(url): Observable<Definitions> {
+        return this.http.get(url)
+         .map(res => res.json())
+         .map(res => res['@context'])
+         .map(hydra => {
+             const definitions = new Definitions();
+             for (let key in hydra) {
+                let value = hydra[key];
+                if (value instanceof Object && typeof value['@id'] !== 'undefined') {
+                    value = value['@id'];
+                }
+                 definitions.add(key, value);
+             }
+             return definitions;
+         });
+
+    }
+
+    setCacheHydraDefinition(definitions: Observable<Definitions>) {
+        definitions.subscribe(res => localStorage.setItem('definition', JSON.stringify(res)));
+    }
+
+    getCacheHydraDefinition() {
+        return localStorage.getItem('definition');
+    }
+
+}
+
+export class Definitions {
```

Comment:
What do you mean ?

Do we still use the defineProperty ? or do we just return an object ?